zk/conn.go

```diff
@@ -81,11 +86,15 @@ type Conn struct {
 	recvTimeout    time.Duration
 	connectTimeout time.Duration
 
+	creds   []authCreds
+	credsMu sync.Mutex // protects server
+
 	sendChan     chan *request
 	requests     map[int32]*request // Xid -> pending request
 	requestsLock sync.Mutex
 	watchers     map[watchPathType][]chan Event
 	watchersLock sync.Mutex
+	closeChan    chan struct{} // channel to tell send loop stop
```

Comment:
I just moved this variable to the struct and kept the name as-is. I do not want to change too much code in this iteration so that it does not obfuscate the patch.